packages/@react-stately/slider/src/useSliderState.ts

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright 2020 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+import {clamp} from '@react-aria/utils';
+import {SliderProps} from '@react-types/slider';
+import {useControlledState} from '@react-stately/utils';
+import {useNumberFormatter} from '@react-aria/i18n';
+import {useRef, useState} from 'react';
+
+export interface SliderState {
+  // Values managed by the slider
+  readonly values: number[],
+  setValue: (index: number, value: number) => void,
+
+  // Whether a specific index is being dragged
+  isDragging: (index: number) => boolean,
+  setDragging: (index: number, dragging: boolean) => void,
+
+  // Currently-focused index
+  readonly focusedIndex: number | undefined,
+  setFocusedIndex: (index: number | undefined) => void,
+
+  // Returns the value offset as a percentage from 0 to 1.
+  getOffsetPercentForIndex: (index: number) => number,
+  getOffsetPercentForValue: (value: number) => number,
+
+  // Returns the string label for the value, per props.formatOptions
+  getValueLabelForIndex: (index: number) => string,
+  getValueLabelForValue: (value: number) => string,
```

Comment:
Maybe `getThumbValueLabel`? I think `getValueLabelForValue` could also be removed, looks like it's not used outside here?